### PR TITLE
enhancement changes are added

### PR DESCRIPTION
**Pull request by anil-intelops**
image

### CODE REVIEW

File: capten/common-pkg/plugins/git/git.go [hunk 1]
```diff
@@ -71,6 +71,15 @@ func (g *GitClient) Commit(msg, name, email string) error {
...
 
+	status, err := w.Status()
+	if err != nil {
+		return err
+	}
+
+	if status.IsClean() {
+		return nil
+	}
+
```
Comment (vramk23) [inline capten/common-pkg/plugins/git/git.go:82]: write debug log, no changes found

File: capten/common-pkg/plugins/argocd/repositories.go [hunk 2]
```diff
@@ -16,6 +16,12 @@ func (a *ArgoCDClient) CreateRepository(ctx context.Context, repo *Repository) (
 	}
 	defer io.Close(conn)
 
+	var repoUpsert bool
+	existingRepo, err := a.GetRepository(ctx, repo.Repo)
+	if existingRepo != nil && err == nil {
+		repoUpsert = true
+	}
+
```
Comment (vramk23) [inline capten/common-pkg/plugins/argocd/repositories.go:24]: check existingRepo, match repo  that it retuined proper value.
first check err !=nil then check existingRepo

File: capten/common-pkg/plugins/argocd/repositories.go [hunk 3]
```diff
@@ -31,6 +37,7 @@ func (a *ArgoCDClient) CreateRepository(ctx context.Context, repo *Repository) (
 				Message: repo.ConnectionState.Message,
 			},
 		},
+		Upsert: repoUpsert,
 	})
```
Comment (vramk23) [inline capten/common-pkg/plugins/argocd/repositories.go:41]: is new project adding gives error if upsert is true ?
if not then, no needed check project exist

File: capten/agent/internal/crossplane/cluster_claims.go [hunk 4]
```diff
@@ -422,3 +441,40 @@ func (h *ClusterClaimSyncHandler) syncClusterClaimsWithDB(clusterClaims []model.
...
 }
+
+func (h *ClusterClaimSyncHandler) checkForClusterUpdate(clusterCliam model.ClusterClaim) bool {
+
+	var statues []string
+	for _, condition := range clusterCliam.Status.Conditions {
+		statues = append(statues, condition.Status)
+	}
+
+	sameStatues := isSameStatues(clusterCache[clusterCliam.Metadata.Name].Statuses, statues)
```
Comment (vramk23) [inline capten/agent/internal/crossplane/cluster_claims.go:452]: its not implemented as discussed.
missing secret handling

Comment (vramk23) [inline capten/agent/internal/crossplane/cluster_claims.go:452]: isSameStatues confusing, have all cluster status check one method.

ClusterCache change to ManagedClusterData
ManagedClusterData - add same attributes
 - add method match

**Review by vramk23 — CHANGES_REQUESTED** [review]

File: capten/common-pkg/plugins/argocd/repositories.go (hunk 3 above)
Comment (anil-intelops) [reply capten/common-pkg/plugins/argocd/repositories.go:41]: yes, we won't get error. removed the project existing check by proving upsert value as true.

File: capten/agent/internal/crossplane/cluster_claims.go (hunk 4 above)
Comment (anil-intelops) [reply capten/agent/internal/crossplane/cluster_claims.go:461]: handling of ManagedClusterData is modified

**Pull request closed without merging**